### PR TITLE
Rename to 'soh'

### PR DESCRIPTION
**Pull request by stappersg**
Two main reasons:
* Dropped the 3 to be ready for HTTP/2
* Avoiding confusion with OpenSSH.  ( We don't want to panic SSH users to update their SSH. And we don't want to keep explaining this project is not next OpenSSH release.)

For reproducing this large change:

vi +/stand README.md
grep -rl ssh3 | grep -v .git/ | xargs sed --in-place -e 's/ssh3/soh/g' grep -rl SSH3 | grep -v .git/ | xargs sed --in-place -e 's/SSH3/SOH/g' find . -name '*ssh3*' -type f
git mv s{sh3,oh}_suite_test.go
git mv integration_tests/s{sh3,oh}_test.go
cd resources/figures/
git mv s{sh3,oh}.png
git mv s{sh3,oh}_oidc.gif
git mv s{sh3,oh}_100ms_rtt.gif
cd -
find . -name '*ssh3*' -type f  # all ssh3 filenames gone?

This change as "point of view from git":
 35 files changed, 351 insertions(+), 351 deletions(-)
 rename integration_tests/{ssh3_test.go => soh_test.go} (90%)
 rename resources/figures/{ssh3.png => soh.png} (100%)
 rename resources/figures/{ssh3_100ms_rtt.gif => soh_100ms_rtt.gif} (100%)
 rename resources/figures/{ssh3_oidc.gif => soh_oidc.gif} (100%)
 rename ssh3_suite_test.go => soh_suite_test.go (90%)

### CODE REVIEW

**Comment by francoismichel** [comment]
Thanks for the PR. I think it might make sense to rebase ith with the upstream that not integrates proxy jump. That will be useful for distros.

**Comment by stappersg** [comment]
On Wed, Jan 10, 2024 at 01:31:20AM -0800, francoismichel wrote:
> Thanks for the PR.

   :-)


> I think it might make sense to rebase ith with the upstream that not integrates proxy jump.

I do understand "rebase", I don't yet get the "with upstream that not
integrates proxy jump". Please elaborate.


Groeten
Geert Stappers
Submitter of merge request #96  (this MR)
-- 
Silence is hard to parse

**Comment by stappersg** [comment]
_rebased_

**Comment by francoismichel** [comment]
Sooo, I think the v0.1.6 is the real good candidate for here, as it integrates a lot of important stuff, including improved logging and proper version negotiation that will both help distro users. Can I ask a last rebase before my green light ? :-)

I can do it myself if needed.

**Comment by stappersg** [comment]
On Wed, Jan 17, 2024 at 12:59:28AM -0800, francoismichel wrote:
> Can I ask a last rebase before my green light ? :-)
> I can do it myself if needed.

That is better. That was and is the idea of the instructions
in the commit message.


Groeten
Geert Stappers


P.S.
Please do the `s/ssh3/soh/g` also in the thesis document
and various URLs  (like the URL to the git repro.)
Go for the long run, think of a future where you don't need
to explain why `soh` comes from `ssh3`.

P.P.S.
If you want to go for `hss`,  HTTP secure shell, just do.
-- 
Silence is hard to parse

**Comment by CodeIter** [comment]
Another suggestion, If you haven't decided yet:

A different name because this is not a new version of openssh.

**SHARQ**  🦈 : **SH**ell **Q**UIC **R**emote **A**ccess .
Client : sharq
Server : sharqd

**Comment by stappersg** [comment]
On Fri, Feb 09, 2024 at 01:25:49AM -0800, Muhammad Amin Boubaker wrote:
> Another suggestion, If you haven't decided yet:
> 
> A different name because this is not a new version of openssh.
> 
> **SHARQ**  🦈 : **SH**ell **Q**UIC **R**emote **A**ccess .
> Client : sharq
> Server : sharqd
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/francoismichel/ssh3/pull/96#issuecomment-1935588045
> You are receiving this because you authored the thread.
> 
> Message ID: ***@***.***>


Thanks for the reminder.

And thanks for removing the comment. #79 would be a better place for
reporting different names because this is not a new version of OpenSSH.


Groeten
Geert Stappers
-- 
Silence is hard to parse

**Comment by stappersg** [comment]
`ping` a.k.a.  _ICMP echo request_
